object/application.go

```diff
@@ -399,7 +400,10 @@ func UpdateApplication(id string, application *Application) (bool, error) {
 		return false, err
 	}
 
+	record := GetRecord(ctx)
 	for _, providerItem := range application.Providers {
+		record.AddReason(fmt.Sprintf("Added provider: %s", providerItem.Name))
```

Comment:
providers can be same as in oldApplication, so no provider was added, but here we will save it as "added provider"

Reply:
if we really need such record, probably we should find diff between old application providers list and new

Reply:
added diff